### ISSUE TITLE
!uptime clarification

### ISSUE THREAD

**Issue by CylonicRaider** [jedevc/botrulez#3]
Should reconnections count as interruptions of bot downtime, should they not, or should it depend on the bot's interna (like, whether it resets relevant state upon reconnections)?
Whilst a re-connect is (pedantically speaking) an interruption of service, it is handled by most libraries to the extent the user code does not notice.

**Comment by CylonicRaider** [comment]
Another question is how bots across multiple rooms (as supported by the major libraries) should be handled — should the uptime be counted by room or in general?
[IMHO, the text suggests the latter.]

**Comment by TauNeutrin0** [comment]
I have used a system whereby the bot responds to !uptime with:
/me has been up since [time]
/me has been online in this room since [time]

This is much clearer and provides more information to the user.

Also, the grammar should be '/me **has been** up since [time]'.

**Comment by CylonicRaider** [comment]
> Also, the grammar should be '/me has been up since [time]'.

This point has been addressed in #4.

**Comment by jedevc** [comment]
My interpretation has always been that reconnections are a regular occurrence and so should not be counted against uptime. My definition of uptime was the amount of time that the running program that runs the bot has been active.

**Comment by jedevc** [comment]
Hm, about the multi room issue.

Judging from program start is tricky. In that case, all BotBot bots would have the same uptime. Which would be incorrect. I'm not quite sure about this. Any ideas?

**Comment by CylonicRaider** [comment]
> In that case, all BotBot bots would have the same uptime.

Good point. So, we should perhaps rule in favor of the in-general-and-per-room solution

**Comment by jedevc** [comment]
Perhaps it would be worth adding a new command? Maybe, `!online` or something like that. Or is that too much effort for such a little thing?

**Comment by CylonicRaider** [comment]
Nah, this is perfectly covered by !uptime.
